clippy_dev/src/fmt.rs

```diff
@@ -260,50 +260,36 @@ fn fmt_syms(update_mode: UpdateMode) {
     );
 }
 
-fn run_rustfmt(clippy: &ClippyInfo, update_mode: UpdateMode) {
+fn run_rustfmt(update_mode: UpdateMode) {
     let mut rustfmt_path = String::from_utf8(run_with_output(
         "rustup which rustfmt",
         Command::new("rustup").args(["which", "rustfmt"]),
     ))
     .expect("invalid rustfmt path");
     rustfmt_path.truncate(rustfmt_path.trim_end().len());
 
-    let mut cargo_path = String::from_utf8(run_with_output(
-        "rustup which cargo",
-        Command::new("rustup").args(["which", "cargo"]),
-    ))
-    .expect("invalid cargo path");
-    cargo_path.truncate(cargo_path.trim_end().len());
-
-    // Start all format jobs first before waiting on the results.
-    let mut children = Vec::with_capacity(16);
-    for &path in &[
-        ".",
-        "clippy_config",
-        "clippy_dev",
-        "clippy_lints",
-        "clippy_lints_internal",
-        "clippy_utils",
-        "rustc_tools_util",
-        "lintcheck",
-    ] {
-        let mut cmd = Command::new(&cargo_path);
-        cmd.current_dir(clippy.path.join(path))
-            .args(["fmt"])
-            .env("RUSTFMT", &rustfmt_path)
-            .stdout(Stdio::null())
-            .stdin(Stdio::null())
-            .stderr(Stdio::piped());
-        if update_mode.is_check() {
-            cmd.arg("--check");
-        }
-        match cmd.spawn() {
-            Ok(x) => children.push(("cargo fmt", x)),
-            Err(ref e) => panic_action(&e, ErrAction::Run, "cargo fmt".as_ref()),
-        }
-    }
+    let args: Vec<_> = WalkDir::new(".")
+        .into_iter()
+        .filter_entry(|e| {
+            // Skip traversing larger directories
+            e.path()
+                .as_os_str()
+                .as_encoded_bytes()
+                .get(2..)
+                .is_none_or(|x| x != "target".as_bytes() && x != ".git".as_bytes())
```

Comment:
At least one Clippy maintainer uses `.jj` which will have the same issue as `.git`:

```suggestion
                .is_none_or(|x| !matches!(x, b"target" | b".git" | b".jj"))
```

Reply:
Not really a problem, but I agree it should be skipped as well.

Reply:
If this is not a colocated git repo, all the git files will be under `.jj/repo/store/git`, which would not be filtered away, hence the importance of including it.

Reply:
Ah. That is indeed a problem.